cloud/wfm/src/task/BaseTask.ts

```diff
@@ -0,0 +1,46 @@
+import { EventEmitter } from 'eventemitter3';
+import { Task, TaskEventData, TaskStatus } from './index';
+
+/**
+ * Base implementation for
+ */
+class BaseTask extends EventEmitter implements Task {
+  protected _status: TaskStatus | number = TaskStatus.pending;
+  protected options: object;
+
+  public set status(to: TaskStatus | number) {
+    const e: TaskEventData<this> = {
+      date: new Date(),
```

Comment:
This is actually just reporting the time itself, so the tracking would be optional and left for client code.

The tracker would listen on these events and register them however it wants to, most likely to external persistent storage, anyway not something we should tackle now.

TL;DR we should design the apis so they can be properly observed by whoever wants to, but not care about why and how for the moment.